src/web/app/svgedit/svgcanvas.ts

```diff
@@ -4660,9 +4630,22 @@ export default $.SvgCanvas = function (container: SVGElement, config: ISVGConfig
     }
 
     // update selection
+    const layers = selectedElements
+      .flatMap((elem) => elem.getAttribute('data-tempgroup') === 'true'
+        ? selectedLayers
+        : LayerHelper.getObjectLayer(elem)?.title
+      );
+
+    // the uniq process is performed `here` to avoid duplicate layer in layer panel,
+    // and remain the selected layers contains information if there are multiple elements in same layer
+    LayerPanelController.setSelectedLayers(layers.filter((layer, index, array) => array.indexOf(layer) === index));
```

Comment:
If the purpose of this line is to make array unique, then how about
`[...new Set(layers)]`